### PR TITLE
Dotnet doc readme for Azure Search (ported from SDK readme)

### PR DESCRIPTION
**Pull request by HeidiSteen**

### CODE REVIEW

**Comment by HeidiSteen** [comment]
#sign-off

File: docs-ref-toc/reference-latest.yml [hunk 1]
```diff
@@ -1813,7 +1813,7 @@
     href: ~/api/overview/azure/latest/search.md
     landingPageType: Service
     items:
-    - name: Cognitive Search
```
Comment (danieljurek) [inline docs-ref-toc/reference-latest.yml:1816]: The `reference-*.yml` files are updated by automation so these changes will be overwritten. Update the `DisplayName` field in this CSV file instead: 

https://github.com/Azure/azure-sdk/blob/main/_data/releases/latest/dotnet-packages.csv#L8

File: docs-ref-toc/reference-latest.yml [hunk 2]
```diff
@@ -1831,7 +1831,7 @@
       landingPageType: Service
     - name: Management
       items:
-      - name: Resource Management - Cognitive Search
```
Comment (danieljurek) [inline docs-ref-toc/reference-latest.yml:1834]: Same as above, update `DisplayName` in this file instead: https://github.com/Azure/azure-sdk/blob/main/_data/releases/latest/dotnet-packages.csv#L135

**Comment by HeidiSteen** [comment]
#sign-off

File: api/overview/azure/latest/search-index.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
 | Reference | Package | Source |
 |---|---|---|
-|[Cognitive Search](search.documents-readme.md)|[Azure.Search.Documents](https://www.nuget.org/packages/Azure.Search.Documents)|[GitHub](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents)|
-|[Resource Management - Cognitive Search](resourcemanager.search-readme.md)|[Azure.ResourceManager.Search](https://www.nuget.org/packages/Azure.ResourceManager.Search)|[GitHub](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.ResourceManager.Search)|
+|[Azure AI Search](search.documents-readme.md)|[Azure.Search.Documents](https://www.nuget.org/packages/Azure.Search.Documents)|[GitHub](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/Azure.Search.Documents)|
```
Comment (danieljurek) [inline api/overview/azure/latest/search-index.md:3]: These changes will likely be overwritten by automation in the next pipeline run. That doesn't affect the merge-ability of this PR. Just FYI.

**Review by danieljurek — APPROVED** [review]

**Pull request merged**